samples/bynder-dialog/src/index.js

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useReducer } from 'react';
+import ReactDOM from 'react-dom';
+import { Button } from '@contentful/forma-36-react-components';
+import { init, locations } from 'contentful-ui-extensions-sdk';
+import '@contentful/forma-36-react-components/dist/styles.css';
+import './index.css';
+import extension from '../extension.json';
+import BynderDialog from './BynderDialog';
+import Thumbnail from './Thumbnail';
+
+function reducer(state, action) {
+  switch (action.type) {
+    case 'add':
```

Comment:
`add` is never dispatched and can easily be replaced with `add-all` anyway;

I'd make `state` an object, with the array being an items property to have an easier time adding additional state elements at a later stage.

In addition, the naming of the actions is very short and does not completely explain what they do; How about `add-all-assets` and `remove-asset` or so?